### ISSUE TITLE
Upgrade from C122 -> C123 on `macOS x64` via `1.65.x - Nightly` (Checking C123)

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#36639]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [ ] `oofiananboodjbbmdelgdommihjbkfag`: HTTPSE
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by LaurenWags** [comment]
Tests above verified with

```
Brave | 1.65.62 Chromium: 123.0.6312.29 (Official Build) nightly (x86_64)
-- | --
Revision | 7760e1c54748b192dd410678c602423529d7a56f
OS | macOS Version 13.6.4 (Build 22G513)
```

**Comment by LaurenWags** [comment]
Upgrade checks in progress (using arm64 .dmg):

```

```

and

```

```

Note, need to be on internal VPN to do the below upgrade checks.

<details>
<summary>C123 --> C123 on-demand upgrade check - InProgress</summary>

1. Install `C123` PR build (arm64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1709854127709629?thread_ts=1709854113.293739&cid=CU5UXM3EC)
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Note, if you want to check deltas, at this point open something like Fiddler Everywhere before continuing.
6. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates-panel.brave.com/sparkle/Brave-Browser/nightly-arm64/appcast.xml`
7. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

8. Confirm you are prompted to "Relaunch" to get update (note, if you click "Relaunch" button you will relaunch and could potentially pick up a newer Nightly that is on the previous Chromium version. Closing and then relaunching via the terminal will not update you again)

SS

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------

</details>
